src/web/components/SharingPermission/BulkAddPermissions.tsx

```diff
@@ -71,19 +77,42 @@ export function BulkAddPermissions({
   const watchDSPChecked = watch('DSPChecked');
   const watchDataProviderChecked = watch('dataProviderChecked');
 
+  const hasCheckedType = useCallback(() => {
+    return (
+      watchPublisherChecked || watchAdvertiserChecked || watchDSPChecked || watchDataProviderChecked
+    );
+  }, [watchPublisherChecked, watchAdvertiserChecked, watchDSPChecked, watchDataProviderChecked]);
+
+  useEffect(() => {
+    setFilteredParticipants(
+      getFilteredParticipantsByType(
```

Comment:
nit: I guess filteredParticipant could managed by useMemo hook? As  we are memoizing the result of an expensive computation

Reply:
hm what would that look like exactly? also where do we draw the line for expensive computation? the `getFilteredParticipantsByType` has linear time complexity, and will comprised of all siteIds which will be quite a long list (at least in future if it isn't already)

Reply:
I've tried:
```
  const filteredParticipantsByType = useMemo(() => {
    return getFilteredParticipantsByType(
      availableParticipants,
      watchPublisherChecked,
      watchAdvertiserChecked,
      watchDSPChecked,
      watchDataProviderChecked
    );
  }, [
    availableParticipants,
    watchPublisherChecked,
    watchAdvertiserChecked,
    watchDSPChecked,
    watchDataProviderChecked,
  ]);

  useEffect(() => {
    console.log('filteredParticipantsByType', filteredParticipantsByType);
    setFilteredParticipants(filteredParticipantsByType);
  }, [filteredParticipantsByType]);
```
but it still seems to hit the useEffect even if `filteredParticipantsByType` returns the same participants (maybe because of a deep equality check?).

tbh I haven't used `useMemo` or `useEffect` that much so open to ideas :)

Reply:
When we have useMemo, we no longer need to manage the state filteredParticipantsByType. So the following could be removed
```
const [filteredParticipants, setFilteredParticipants] = useState(availableParticipants);

useEffect(() => {
    console.log('filteredParticipantsByType', filteredParticipantsByType);
    setFilteredParticipants(filteredParticipantsByType);
  }, [filteredParticipantsByType]);
```

and it will be 
```
  const filteredParticipants = useMemo(() => {
    return getFilteredParticipantsByType(
      availableParticipants,
      watchPublisherChecked,
      watchAdvertiserChecked,
      watchDSPChecked,
      watchDataProviderChecked
    );
  }, [
    availableParticipants,
    watchPublisherChecked,
    watchAdvertiserChecked,
    watchDSPChecked,
    watchDataProviderChecked,
  ]);

          {filteredParticipants.map((p) => (
            <tr key={p.id}>
              <ParticipantItemSimple participant={p} />
            </tr>
          ))}
```

https://stackoverflow.com/questions/56347639/react-useeffect-vs-usememo-vs-usestate

Reply:
ahh gotcha! so this will ensure that React will return the same value on subsequent renders given that the dependencies haven't changed. though as I understand it, this same behaviour can be obtained by using `useState` and `useEffect` together, although using `useMemo` makes it easier to understand. Also `useMemo` runs during the render while `useState` causes a re-render and `useEffect` occurs after each render.

done